hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -1360,6 +1369,57 @@ private void nodeSpaceStatistics(Map<String, String> nodeStatics) {
     nodeStatics.put(SpaceStatistics.NON_SCM_USED.getLabel(), nonScmUsed);
   }
 
+  private void nodeReadOnlyStatistics(Map<String, String> nodeStatics) {
+    List<DatanodeInfo> allNodes = getAllNodes();
+    long blockSize = (long) conf.getStorageSize(
+        OzoneConfigKeys.OZONE_SCM_BLOCK_SIZE,
+        OzoneConfigKeys.OZONE_SCM_BLOCK_SIZE_DEFAULT,
+        StorageUnit.BYTES);
+    long minRatisVolumeSizeBytes = (long) conf.getStorageSize(
+        ScmConfigKeys.OZONE_DATANODE_RATIS_VOLUME_FREE_SPACE_MIN,
+        ScmConfigKeys.OZONE_DATANODE_RATIS_VOLUME_FREE_SPACE_MIN_DEFAULT,
+        StorageUnit.BYTES);
+    long containerSize = (long) conf.getStorageSize(
+        ScmConfigKeys.OZONE_SCM_CONTAINER_SIZE,
+        ScmConfigKeys.OZONE_SCM_CONTAINER_SIZE_DEFAULT,
+        StorageUnit.BYTES);
+
+    int readOnlyCount = (int) allNodes.parallelStream()
+        .filter(dnInfo -> {
+          boolean hasWritableContainer = hasContainerWithSpace(dnInfo, blockSize, containerSize);
+          if (hasWritableContainer) {
+            return false;
+          }
+
+          boolean canAllocateNewContainers = SCMCommonPlacementPolicy.hasEnoughSpace(
+              dnInfo, minRatisVolumeSizeBytes, containerSize, conf);
+          return !canAllocateNewContainers;
+        })
```

Comment:
Would it make sense to check `hasEnoughSpace` first?  `hasContainerWithSpace` seems to be the more expensive operation.

Also, I think this would be more easier to understand as:

```java
  .filter(dn -> !hasEnoughSpace(dn, minRatisVolumeSizeBytes, containerSize, conf)
      && !hasContainerWithSpace(dn, blockSize, containerSize))
```